core/src/net/cidr/v4.rs

```diff
@@ -45,6 +58,34 @@ impl Default for Ipv4Cidr {
     }
 }
 
+#[derive(Clone, Debug)]
+pub struct Ipv4CidrIterator {
+    next: Option<u32>,
+    end: u32,
+}
+
+impl Iterator for Ipv4CidrIterator {
+    type Item = Ipv4Addr;
+
+    fn next(&mut self) -> Option<Ipv4Addr> {
+        let next = self.next?;
+        self.next = if next == self.end {
+            None
+        } else {
+            Some(next + 1)
+        };
+        Some(next.into())
+    }
+}
+
+impl IntoIterator for &'_ Ipv4Cidr {
```

Comment:
is this abusing the ownership? technically when you use `into` it suppose to take ownership of `self`, but here, it's an impl on the borrow.

Reply:
this came from the idea used in Vec: https://doc.rust-lang.org/src/alloc/vec.rs.html#2026. I'm pushing up an update that has a clearer iter/into_iter uses akin to https://github.com/stbuehler/rust-cidr. Let's discuss tmrw.